### ISSUE TITLE
Autoclosing tags for XML (and XHTML)

### ISSUE THREAD

**Issue by shmargum** [k0kubun/hamlit#92]
Currently autoclosing tags like link will render like so:
`<link href="some.url.com">`

But for XML (and XHTML) they should be closed like:
`<link href="some.url.com" />`

I have both HTML and XML templates, and cannot set the `:format` option globally.

In haml, the link tags treated correctly in each format.

**Comment by k0kubun** [comment]
I couldn't reproduce it.

```rb
[4] 2.3.1-p112 (main)> eval Hamlit::Engine.new(format: :html).call("%link{ href: 'some.url.com' }")
=> "<link href='some.url.com'>\n"
[5] 2.3.1-p112 (main)> eval Hamlit::Engine.new(format: :xhtml).call("%link{ href: 'some.url.com' }")
=> "<link href='some.url.com' />\n"
[6] 2.3.1-p112 (main)> Hamlit::Engine.options[:format] = :html
=> :html
[7] 2.3.1-p112 (main)> eval Hamlit::Engine.new.call("%link{ href: 'some.url.com' }")
=> "<link href='some.url.com'>\n"
[8] 2.3.1-p112 (main)> Hamlit::Engine.options[:format] = :xhtml
=> :xhtml
[9] 2.3.1-p112 (main)> eval Hamlit::Engine.new.call("%link{ href: 'some.url.com' }")
=> "<link href='some.url.com' />\n"
[10] 2.3.1-p112 (main)> Hamlit::VERSION
=> "2.6.0"
```

Please write more details.

- How did you specify `:format` option?
- Do you use Hamlit in Rails or Sinatra?

Could you try https://github.com/k0kubun/hamlit/blob/master/REFERENCE.md#set-options-for-rails or https://github.com/k0kubun/hamlit/blob/master/REFERENCE.md#set-options-for-sinatra?

**Comment by k0kubun** [comment]
Ah, sorry, you can set `:format` option globally but you don't want to.

**Comment by k0kubun** [comment]
Again, do you use Rails? When using haml, how did you tell haml which format you use? By filename, like "foo.html.haml" and "bar.xhtml.haml"?

**Comment by shmargum** [comment]
Yes I'm using Rails.
The file is an XML file like "foo.xml.haml";
`render file: 'foo.xml'`
I did not do anything special for haml to recognize it as an xhtml format.   I tried passing in format and content_type options without success.

**Comment by k0kubun** [comment]
Fixed this issue on Hamlit v2.7.4. Could you try it?

**Comment by shmargum** [comment]
Thanks, seems to replicate the behavior correctly. 👏

**Comment by shmargum** [comment]
Sorry, I'm having the following issue when upgrading:
`/Users/maximivanov1/huffpost-mobileweb/config/initializers/hamlit.rb:2:in `<top (required)>': undefined method `set_options' for Hamlit::RailsTemplate:Class (NoMethodError)`

**Comment by k0kubun** [comment]
Sorry, resolved the problem in Hamlit v2.7.5 and yanked v2.7.4. Could you try it again?

**Comment by shmargum** [comment]
perfect, thanks ❤️
